packages/marshal/src/rankOrder.js

```diff
@@ -55,15 +71,20 @@ export const trivialComparator = (left, right) =>
 const passStyleRanks = /** @type {PassStyleRanksRecord} */ (
   fromEntries(
     entries(passStylePrefixes)
-      // Sort entries by ascending prefix.
-      .sort(([_leftStyle, leftPrefixes], [_rightStyle, rightPrefixes]) => {
-        return trivialComparator(leftPrefixes, rightPrefixes);
+      // Sort entries by ascending prefix, assuming that all prefixes are
+      // limited to the Basic Multilingual Plane (U+0000 through U+FFFF) and
+      // thus contain only code units that are equivalent to code points.
+      // In practice, they are entirely printable ASCII
+      // (0x20 SPACE through 0x7E TILDE).
+      .sort(([_aStyle, aPrefixes], [_bStyle, bPrefixes]) => {
+        // eslint-disable-next-line no-nested-ternary
+        return aPrefixes < bPrefixes ? -1 : aPrefixes > bPrefixes ? 1 : 0;
       })
       .map(([passStyle, prefixes], index) => {
-        // Cover all strings that start with any character in `prefixes`,
-        // verifying that it is sorted so that is
+        // Verify that `prefixes` is sorted, and cover all strings that start
+        // with any of its characters, i.e.
         // all s such that prefixes.at(0) ≤ s < successor(prefixes.at(-1)).
-        prefixes === [...prefixes].sort().join('') ||
+        prefixes === prefixes.split('').sort().join('') ||
```

Comment:
Is `prefixes` an array of strings or a string?

Reply:
It's a string from [`passStylePrefixes`](https://github.com/endojs/endo/blob/db9e48f24fd3b532c8478e5cfc0cf9720fc13fcb/packages/marshal/src/encodePassable.js#L867-L884), usually one character like "s" or "?", but two characters for copyArray "[^" (which also covers "\" and "]") and bigint "np" (which also covers "o").